torchmetrics/regression/explained_variance.py

```diff
@@ -112,14 +108,22 @@ def update(self, preds: torch.Tensor, target: torch.Tensor):
             preds: Predictions from model
             target: Ground truth values
         """
-        preds, target = _explained_variance_update(preds, target)
-        self.y_pred.append(preds)
-        self.y.append(target)
+        n_obs, sum_error, sum_squared_error, sum_target, sum_squared_target = _explained_variance_update(preds, target)
+        self.n_obs = self.n_obs + n_obs
+        self.sum_error = self.sum_error + sum_error
+        self.sum_squared_error = self.sum_squared_error + sum_squared_error
+        self.sum_target = self.sum_target + sum_target
+        self.sum_squared_target = self.sum_squared_target + sum_squared_target
```

Comment:
```suggestion
        self.n_obs += n_obs
        self.sum_error += sum_error
        self.sum_squared_error += sum_squared_error
        self.sum_target += sum_target
        self.sum_squared_target += sum_squared_target
```

Reply:
Using += was actually one of the problems with the tests earlier. It doesn't work when target and preds are of shape [batch_size, n] with n > 1 because the default in the self.add_state calls in __init__ is a tensor of size [].  I found that spelling out the update as `a = a+b` instead was a workaround, maybe there's a cleaner fix for it though.

Updated the changelog.

Reply:
alright sounds good to me :]